p2p/discovery/ping.go

```diff
@@ -38,6 +40,7 @@ func (p *protocol) newPingRequestHandler() func(msg server.Message) []byte {
 		}
 
 		plogger.Debug("Sending pong message")
+		plogger.Warning("Sending pong message")
 		return payload
 	}
 }
```

Comment:
we should probably check if a peer `NeedsPing` after we got pinged from him, this is used to make sure the pinger with his provided identity actually owns the IP from the incoming ping, otherwise he could use it to create a reflected DoS attack and make many peers try and reach specific address, so we check this before we save an address in the addressbook.

Reply:
This makes sense. If the peer who's sending the ping `NeedsPing` within the `AddressBook`, what should the desired behavior be? We should not issue another ping request right, because that would risk an infinite loop of pings. Should the desired behavior then be to not reply to the ping request?

Reply:
I see your point, we might need to check this only before sending a GetAdderss\Addresses message because these are bigger messages, anyway this was inspired from ethereum's `Bond` mechanism, please take a look here -
https://github.com/ethereum/go-ethereum/blob/4adf0b9187060539e8c0dd9eda2ad5a7d3c53cac/p2p/discover/v4_udp.go#L570

Reply:
We currently check whether the Peer needs ping before the GetAddresses. If the peer needs the ping, then we send a ping and without a successful response, we don't respond.